### PR TITLE
Fix golangcli-lint issue

### PR DESCRIPTION
**Pull request by erkanerol**
Signed-off-by: Erkan Erol <eerol@redhat.com>

We are hitting https://github.com/golangci/golangci-lint-action/issues/135 in our checks. This PR disables some functionalities of the linter as a workaround shared here: https://github.com/golangci/golangci-lint-action/issues/135#issuecomment-791775296

**Reviewer Checklist**

- [x] PR Message
- [x] Commit Messages
- [x] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by nunnatsa** [comment]
pr-sanity action didn't run. It seems that the file indentation is wrong.

File: .github/workflows/pr-sanity.yaml [hunk 1]
```diff
@@ -36,6 +36,8 @@ jobs:
 
       - name: Run golangci-lint
         uses: golangci/golangci-lint-action@v2.5.2
+          skip-go-installation: true
+          skip-pkg-cache: true
```
Comment (nunnatsa) [inline .github/workflows/pr-sanity.yaml:40]: Wrong indentation.

**Review by nunnatsa — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-sanity.yaml (hunk 1 above)
Comment (erkanerol) [reply .github/workflows/pr-sanity.yaml:41]: sorry :( fixed.

**Comment by nunnatsa** [comment]
Failed because of linter timeout.

Let's take out the temp `kubevirt` directory. Maybe also the `tests` directory.

**Comment by tiraboschi** [comment]
> Failed because of linter timeout.
> 
> Let's take out the temp `kubevirt` directory. Maybe also the `tests` directory.

on my machine it's running in 3 seconds:
```
stirabos@t14s:~/go/src/github.com/kubevirt/hyperconverged-cluster-operator$ time golangci-lint run
real	0m0,936s
user	0m3,169s
sys	0m0,418s
```
while the github action is failing due to timeout after 98s

**Comment by erkanerol** [comment]
I increased the timeout from 1m to 5m. I am checking the stability and possible configurations.

**Comment by nunnatsa** [comment]
> on my machine it's running in 1 second:
> 
> ```
> stirabos@t14s:~/go/src/github.com/kubevirt/hyperconverged-cluster-operator$ time golangci-lint run
> real	0m0,936s
> user	0m3,169s
> sys	0m0,418s
> ```

Local caching?

**Comment by nunnatsa** [comment]
> I increased the timeout from 1m to 5m. I am checking the stability and possible configurations.

I still think we don't need to check the whole project.

**Comment by erkanerol** [comment]
> > I increased the timeout from 1m to 5m. I am checking the stability and possible configurations.
> 
> I still think we don't need to check the whole project.

According to my tests, it doesn't check the kubevirt directory.  However, it does check `tests`. I prefer checking tests as well since we need to ensure the quality of the tests as well.

**Review by tiraboschi — APPROVED** [review]

**Pull request merged**